java-runtime-builder-app/src/main/java/com/google/cloud/runtimes/builder/config/domain/BuildContext.java

```diff
@@ -146,22 +146,17 @@ public void writeDockerResources() throws IOException {
   }
 
   private void writeDockerFile() throws IOException {
-    Path dockerFilePath = workspaceDir.resolve(DOCKERFILE_NAME);
-
     // fail loudly if a Dockerfile already exists
-    if (Files.exists(dockerFilePath)) {
-      throw new IllegalStateException("Custom Dockerfiles are not supported. If you wish to use a "
-          + "custom Dockerfile, consider using runtime: custom. Otherwise, remove the Dockerfile "
-          + "from the root of your sources to continue.");
-    }
-
     for (String dir : EXISITING_DOCKER_DIRS) {
       if (Files.exists(workspaceDir.resolve(dir + DOCKERFILE_NAME))) {
-        throw new IllegalStateException("There is an unexpected Dockerfile in " + dir
-            + ". Please remove it to continue.");
+        throw new IllegalStateException("Custom Dockerfiles aren't supported. If you wish to use a "
+            + "custom Dockerfile, consider using runtime: custom. Otherwise, remove the Dockerfile "
+            + "from " + dir + " to continue.");
```

Comment:
In the case of the root dir, this will just say `from to continue`.
maybe something like: `"from the " + (dir == "" ? "root" : dir) + " directory of your sources to continue"`.